process/block/processedMb/processedMiniBlocks.go

```diff
@@ -0,0 +1,135 @@
+package processedMb
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/process/block/bootstrapStorage"
+)
+
+var log = logger.GetOrCreate("process/processedMb")
+
+// MiniBlockHashes will keep a list of miniblock hashes as keys in a map for easy access
+type MiniBlockHashes map[string]struct{}
+
+// ProcessedMiniBlockTracker is used to store all processed mini blocks hashes grouped by a metahash
+type ProcessedMiniBlockTracker struct {
+	processedMiniBlocks    map[string]MiniBlockHashes
+	mutProcessedMiniBlocks sync.RWMutex
+}
+
+// NewProcessedMiniBlocks will create a complex type of processedMb
+func NewProcessedMiniBlocks() *ProcessedMiniBlockTracker {
+	return &ProcessedMiniBlockTracker{
+		processedMiniBlocks: make(map[string]MiniBlockHashes),
+	}
+}
+
+// AddMiniBlockHash will add a miniblock hash
+func (pmb *ProcessedMiniBlockTracker) AddMiniBlockHash(metaBlockHash string, miniBlockHash string) {
+	pmb.mutProcessedMiniBlocks.Lock()
+	defer pmb.mutProcessedMiniBlocks.Unlock()
+
+	miniBlocksProcessed, ok := pmb.processedMiniBlocks[metaBlockHash]
+	if !ok {
+		miniBlocksProcessed := make(MiniBlockHashes)
+		miniBlocksProcessed[miniBlockHash] = struct{}{}
+		pmb.processedMiniBlocks[metaBlockHash] = miniBlocksProcessed
+
+		return
+	}
+
+	miniBlocksProcessed[miniBlockHash] = struct{}{}
+}
+
+// RemoveMetaBlockHash will remove a meta block hash
+func (pmb *ProcessedMiniBlockTracker) RemoveMetaBlockHash(metaBlockHash string) {
+	pmb.mutProcessedMiniBlocks.Lock()
+	delete(pmb.processedMiniBlocks, metaBlockHash)
+	pmb.mutProcessedMiniBlocks.Unlock()
+}
+
+// RemoveMiniBlockHash will remove a mini block hash
+func (pmb *ProcessedMiniBlockTracker) RemoveMiniBlockHash(miniBlockHash string) {
+	pmb.mutProcessedMiniBlocks.Lock()
+	for metaHash, miniBlocksProcessed := range pmb.processedMiniBlocks {
+		_, isProcessed := miniBlocksProcessed[miniBlockHash]
+		if isProcessed {
```

Comment:
it's safe to delete directly, without checking if it exists before

Reply:
done.